frontend/src/components/common/SnackBar/SnackBar.tsx

```diff
@@ -0,0 +1,27 @@
+import ReactDOM from 'react-dom'
+
+import { useRecoilValue } from 'recoil';
+import { snackBarState } from '@/store/snackBarState';
+
+import * as S from '@/components/common/SnackBar/SnackBar.styles';
+
+
+const SnackBar = () => {
+    const snackBar = document.getElementById('snack-bar');
+    const {isOpen, message} = useRecoilValue(snackBarState);
+
+    if(!snackBar){
+        throw new Error('스낵바를 찾지 못하였습니다');
+    }
+   
+   return isOpen ? ReactDOM.createPortal(
```

Comment:
여기서 snackbar 부분과 portal부분을 분리하는것이 어떨까요? 만약 이 컴포넌트에서 isOpen이 변경된다면 dom도 다시 찾고 createPortal도 다시 실행할것같아서 비용이 많이 들것같네요!



```suggestion
const SnackBarPortal = () => {
    const snackBar = document.getElementById('snack-bar');
    return React.createPortal(children)

const SnackBar = () => {

   const {isOpen, message} = useRecoilValue(snackbarState);

    return (
        <SnackbarPortal>
             {isOpen && component}
        </SnackbarPortal>
    )
}
```

이런식으로 하면 좋을것같아요!

Reply:
수정했습니다!